File: app/src/main/java/com/example/solar_alarm/Data/AlarmDao.java
package com.example.solar_alarm.Data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface AlarmDao {
    // Create
    @Insert
    void insert(Alarm alarm);

    // Read
    @Query("SELECT * FROM alarm_table ORDER BY created ASC")
    LiveData<List<Alarm>> getAlarms();

    // Update
    @Update
    void update(Alarm alarm);

    //@Query("DELETE FROM alarm_table WHERE alarmId = :alarmId")
    @Delete
    void delete(Alarm alarm);

    @Query("DELETE FROM alarm_table")
    void deleteAll();
}

File: app/src/main/java/com/example/solar_alarm/Data/AlarmDao.kt
package com.example.solar_alarm.Data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface AlarmDao {
    // Create
    @Insert
    fun insert(alarm: Alarm?)

    // Read all
    @get:Query("SELECT * FROM alarm_table ORDER BY created ASC")
    val alarms: LiveData<List<Alarm?>?>?

    // Update
    @Update
    fun update(alarm: Alarm?)

    //@Query("DELETE FROM alarm_table WHERE alarmId = :alarmId")
    @Delete
    fun delete(alarm: Alarm?)

    @Query("DELETE FROM alarm_table")
    fun deleteAll()
}